### PR TITLE
Separate restore code from the rest of odin

### PR DESCRIPTION
**Pull request by ifosch**
Trying to define a clearer CLI interface for odin,
this PR :

* Removes debugging leftovers
* Moves tests to a separate package
* Separates restore code from the rest
* Adapts tests to separated code
* Makes the command to use new code

So the point here, is to keep the functionality of restoring from a snapshot to a new instance in the VPC, specifying VPC security groups. Something like this:
```bash
odin instance restore -f prod-db -g sg-111111 -n staging-subnet stag-db
```

For review, I would recommend to review commits separately.

### CODE REVIEW

**Comment by ifosch** [comment]
[B] WIP

File: mcc/odin/odin/instance_restore_test.go [hunk 1]
```diff
@@ -97,10 +85,10 @@ func TestGetRestoreDBInstanceFromDBSnapshotInput(t *testing.T) {
 						)
 					}
 				} else {
-					if !equalsRestoreDBInstanceFromDBSnapshotInput(
+					if diff := deep.Equal(
```
Comment (yaneq) [inline mcc/odin/odin/instance_restore_test.go]: i wonder what this is doing

File: mcc/odin/cmd/instance_restore.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/Devex/spaceflight/mcc/odin/odin"
+)
```
Comment (yaneq) [inline mcc/odin/cmd/instance_restore.go:11]: empty lines intended?

**Comment by yaneq** [comment]
+1 i think

File: mcc/odin/cmd/instance_restore.go (hunk 2 above)
Comment (ifosch) [reply mcc/odin/cmd/instance_restore.go:11]: yes, to separate stdlib imports from 3rd parties imports from our own libraries imports :)

File: mcc/odin/odin/instance_restore_test.go (hunk 1 above)
Comment (miguelbernadi) [reply mcc/odin/odin/instance_restore_test.go]: It is initializing the `diff` variable to the result of a function called `deep.Equal` that compares structures field by field. Right afterwards checks that diff is true, so the structures really are different, to report a test error.

**Comment by dsager** [comment]
+1
:see_no_evil:

**Comment by christofdamian** [comment]
+1

**Pull request merged**